fix(useRefEffet): Make IE11 friendly

ling1726 opened the pull request:
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #19181
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Previous useRefEffect hook used `Object.assign` which is not supported
by IE11 without polyfills.

Modify the hook similar to remove `Object.assign`

#### Focus areas to test

(optional)

fabricteam commented:
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/b0f4787d7d20447853191dcb59e848ff12615c46'>b0f4787d7d20447853191dcb59e848ff12615c46</a></sub>
<div id="bundleSizeComment9423"/>

fabricteam commented:
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Avatar-mount_base.html">944</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Avatar-mount.html">931</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/BaseButton-mount_base.html">972</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/BaseButton-mount.html">1026</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Breadcrumb-mount_base.html">2807</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Breadcrumb-mount.html">2827</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ButtonNext</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ButtonNext-mount_base.html">493</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ButtonNext-mount.html">472</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Checkbox-mount_base.html">1740</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Checkbox-mount.html">1708</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/CheckboxBase-mount_base.html">1527</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/CheckboxBase-mount.html">1487</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ChoiceGroup-mount_base.html">5620</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ChoiceGroup-mount.html">5378</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ComboBox-mount_base.html">1074</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ComboBox-mount.html">1092</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/CommandBar-mount_base.html">11629</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/CommandBar-mount.html">11281</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ContextualMenu-mount_base.html">6823</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ContextualMenu-mount.html">6844</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/DefaultButton-mount_base.html">1280</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/DefaultButton-mount.html">1292</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/DetailsRow-mount_base.html">4194</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/DetailsRow-mount.html">4177</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/DetailsRowFast-mount_base.html">4229</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/DetailsRowFast-mount.html">4164</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/DetailsRowNoStyles-mount_base.html">3946</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/DetailsRowNoStyles-mount.html">4006</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Dialog-mount_base.html">2327</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Dialog-mount.html">2304</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/DocumentCardTitle-mount_base.html">151</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/DocumentCardTitle-mount.html">157</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Dropdown-mount_base.html">3771</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Dropdown-mount.html">4032</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderNext</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/FluentProviderNext-mount_base.html">7569</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/FluentProviderNext-mount.html">7672</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/FocusTrapZone-mount_base.html">1949</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/FocusTrapZone-mount.html">1947</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/FocusZone-mount_base.html">1958</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/FocusZone-mount.html">1926</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/IconButton-mount_base.html">1971</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/IconButton-mount.html">1966</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Label-mount_base.html">371</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Label-mount.html">374</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Layer-mount_base.html">2199</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Layer-mount.html">2080</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Link-mount_base.html">540</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Link-mount.html">535</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/MakeStyles-mount_base.html">2073</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/MakeStyles-mount.html">2022</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/MenuButton-mount_base.html">1743</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/MenuButton-mount.html">1703</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/MessageBar-mount_base.html">2188</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/MessageBar-mount.html">2214</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Nav-mount_base.html">3745</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Nav-mount.html">3717</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/OverflowSet-mount_base.html">1165</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/OverflowSet-mount.html">1138</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Panel-mount_base.html">2310</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Panel-mount.html">2226</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Persona-mount_base.html">945</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Persona-mount.html">942</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Pivot-mount_base.html">1601</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Pivot-mount.html">1575</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/PrimaryButton-mount_base.html">1453</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/PrimaryButton-mount.html">1461</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Rating-mount_base.html">9149</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Rating-mount.html">9112</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/SearchBox-mount_base.html">1560</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/SearchBox-mount.html">1482</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Shimmer-mount_base.html">2982</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Shimmer-mount.html">2968</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Slider-mount_base.html">2205</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Slider-mount.html">2261</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/SpinButton-mount_base.html">5632</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/SpinButton-mount.html">5613</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Spinner-mount_base.html">475</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Spinner-mount.html">488</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/SplitButton-mount_base.html">3505</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/SplitButton-mount.html">3574</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Stack-mount_base.html">566</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Stack-mount.html">543</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/StackWithIntrinsicChildren-mount_base.html">1832</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/StackWithIntrinsicChildren-mount.html">1800</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/StackWithTextChildren-mount_base.html">5349</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/StackWithTextChildren-mount.html">5284</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/SwatchColorPicker-mount_base.html">11371</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/SwatchColorPicker-mount.html">11372</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Tabs</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Tabs-mount_base.html">1575</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Tabs-mount.html">1553</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/TagPicker-mount_base.html">2933</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/TagPicker-mount.html">2933</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/TeachingBubble-mount_base.html">60584</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/TeachingBubble-mount.html">59031</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Text-mount_base.html">486</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Text-mount.html">480</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/TextField-mount_base.html">1590</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/TextField-mount.html">1570</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ThemeProvider-mount_base.html">1293</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ThemeProvider-mount.html">1318</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ThemeProvider-virtual-rerender_base.html">643</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/ThemeProvider-virtual-rerender.html">689</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Toggle-mount_base.html">930</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/Toggle-mount.html">949</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/buttonNative-mount_base.html">126</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test/buttonNative-mount.html">132</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

## Perf Analysis (`@fluentui/react-northstar`)
<details><summary>Perf tests with no regressions</summary>

Scenario | Current PR Ticks | Baseline Ticks | Ratio
:--- | ---:| ---:| ---:
AttachmentMinimalPerf.default | [191](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AttachmentMinimalPerf.default.html) | [170](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AttachmentMinimalPerf.default_base.html) | 1.12:1
AvatarMinimalPerf.default | [234](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AvatarMinimalPerf.default.html) | [214](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AvatarMinimalPerf.default_base.html) | 1.09:1
ButtonMinimalPerf.default | [220](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ButtonMinimalPerf.default.html) | [203](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ButtonMinimalPerf.default_base.html) | 1.08:1
ButtonSlotsPerf.default | [648](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ButtonSlotsPerf.default.html) | [601](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ButtonSlotsPerf.default_base.html) | 1.08:1
TreeWith60ListItems.default | [208](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TreeWith60ListItems.default.html) | [192](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TreeWith60ListItems.default_base.html) | 1.08:1
AlertMinimalPerf.default | [323](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AlertMinimalPerf.default.html) | [303](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AlertMinimalPerf.default_base.html) | 1.07:1
StatusMinimalPerf.default | [791](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/StatusMinimalPerf.default.html) | [742](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/StatusMinimalPerf.default_base.html) | 1.07:1
FlexMinimalPerf.default | [327](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/FlexMinimalPerf.default.html) | [309](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/FlexMinimalPerf.default_base.html) | 1.06:1
HeaderSlotsPerf.default | [898](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/HeaderSlotsPerf.default.html) | [846](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/HeaderSlotsPerf.default_base.html) | 1.06:1
DialogMinimalPerf.default | [853](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/DialogMinimalPerf.default.html) | [815](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/DialogMinimalPerf.default_base.html) | 1.05:1
FormMinimalPerf.default | [485](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/FormMinimalPerf.default.html) | [463](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/FormMinimalPerf.default_base.html) | 1.05:1
LoaderMinimalPerf.default | [786](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/LoaderMinimalPerf.default.html) | [748](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/LoaderMinimalPerf.default_base.html) | 1.05:1
TableMinimalPerf.default | [463](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TableMinimalPerf.default.html) | [441](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TableMinimalPerf.default_base.html) | 1.05:1
ButtonOverridesMissPerf.default | [1931](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ButtonOverridesMissPerf.default.html) | [1848](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ButtonOverridesMissPerf.default_base.html) | 1.04:1
CarouselMinimalPerf.default | [543](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/CarouselMinimalPerf.default.html) | [524](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/CarouselMinimalPerf.default_base.html) | 1.04:1
ListNestedPerf.default | [642](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ListNestedPerf.default.html) | [616](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ListNestedPerf.default_base.html) | 1.04:1
AttachmentSlotsPerf.default | [1197](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AttachmentSlotsPerf.default.html) | [1159](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AttachmentSlotsPerf.default_base.html) | 1.03:1
CardMinimalPerf.default | [644](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/CardMinimalPerf.default.html) | [623](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/CardMinimalPerf.default_base.html) | 1.03:1
ChatDuplicateMessagesPerf.default | [340](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ChatDuplicateMessagesPerf.default.html) | [330](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ChatDuplicateMessagesPerf.default_base.html) | 1.03:1
ItemLayoutMinimalPerf.default | [1417](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ItemLayoutMinimalPerf.default.html) | [1375](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ItemLayoutMinimalPerf.default_base.html) | 1.03:1
ListCommonPerf.default | [740](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ListCommonPerf.default.html) | [717](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ListCommonPerf.default_base.html) | 1.03:1
RosterPerf.default | [1308](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/RosterPerf.default.html) | [1276](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/RosterPerf.default_base.html) | 1.03:1
PopupMinimalPerf.default | [666](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/PopupMinimalPerf.default.html) | [649](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/PopupMinimalPerf.default_base.html) | 1.03:1
ProviderMinimalPerf.default | [1127](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ProviderMinimalPerf.default.html) | [1099](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ProviderMinimalPerf.default_base.html) | 1.03:1
RadioGroupMinimalPerf.default | [517](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/RadioGroupMinimalPerf.default.html) | [503](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/RadioGroupMinimalPerf.default_base.html) | 1.03:1
SegmentMinimalPerf.default | [403](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/SegmentMinimalPerf.default.html) | [392](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/SegmentMinimalPerf.default_base.html) | 1.03:1
SkeletonMinimalPerf.default | [431](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/SkeletonMinimalPerf.default.html) | [420](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/SkeletonMinimalPerf.default_base.html) | 1.03:1
DividerMinimalPerf.default | [405](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/DividerMinimalPerf.default.html) | [399](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/DividerMinimalPerf.default_base.html) | 1.02:1
DropdownManyItemsPerf.default | [820](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/DropdownManyItemsPerf.default.html) | [803](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/DropdownManyItemsPerf.default_base.html) | 1.02:1
LayoutMinimalPerf.default | [410](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/LayoutMinimalPerf.default.html) | [400](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/LayoutMinimalPerf.default_base.html) | 1.02:1
RefMinimalPerf.default | [259](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/RefMinimalPerf.default.html) | [255](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/RefMinimalPerf.default_base.html) | 1.02:1
TextMinimalPerf.default | [390](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TextMinimalPerf.default.html) | [383](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TextMinimalPerf.default_base.html) | 1.02:1
AnimationMinimalPerf.default | [467](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AnimationMinimalPerf.default.html) | [462](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AnimationMinimalPerf.default_base.html) | 1.01:1
ChatMinimalPerf.default | [731](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ChatMinimalPerf.default.html) | [727](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ChatMinimalPerf.default_base.html) | 1.01:1
DropdownMinimalPerf.default | [3370](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/DropdownMinimalPerf.default.html) | [3321](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/DropdownMinimalPerf.default_base.html) | 1.01:1
GridMinimalPerf.default | [384](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/GridMinimalPerf.default.html) | [382](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/GridMinimalPerf.default_base.html) | 1.01:1
MenuButtonMinimalPerf.default | [1869](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/MenuButtonMinimalPerf.default.html) | [1845](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/MenuButtonMinimalPerf.default_base.html) | 1.01:1
PortalMinimalPerf.default | [187](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/PortalMinimalPerf.default.html) | [186](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/PortalMinimalPerf.default_base.html) | 1.01:1
SplitButtonMinimalPerf.default | [4345](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/SplitButtonMinimalPerf.default.html) | [4290](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/SplitButtonMinimalPerf.default_base.html) | 1.01:1
CustomToolbarPrototype.default | [4255](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/CustomToolbarPrototype.default.html) | [4217](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/CustomToolbarPrototype.default_base.html) | 1.01:1
TreeMinimalPerf.default | [923](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TreeMinimalPerf.default.html) | [911](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TreeMinimalPerf.default_base.html) | 1.01:1
VideoMinimalPerf.default | [722](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/VideoMinimalPerf.default.html) | [712](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/VideoMinimalPerf.default_base.html) | 1.01:1
BoxMinimalPerf.default | [387](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/BoxMinimalPerf.default.html) | [387](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/BoxMinimalPerf.default_base.html) | 1:1
ImageMinimalPerf.default | [417](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ImageMinimalPerf.default.html) | [419](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ImageMinimalPerf.default_base.html) | 1:1
LabelMinimalPerf.default | [455](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/LabelMinimalPerf.default.html) | [457](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/LabelMinimalPerf.default_base.html) | 1:1
ReactionMinimalPerf.default | [437](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ReactionMinimalPerf.default.html) | [437](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ReactionMinimalPerf.default_base.html) | 1:1
TableManyItemsPerf.default | [2234](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TableManyItemsPerf.default.html) | [2228](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TableManyItemsPerf.default_base.html) | 1:1
ToolbarMinimalPerf.default | [1089](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ToolbarMinimalPerf.default.html) | [1085](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ToolbarMinimalPerf.default_base.html) | 1:1
TooltipMinimalPerf.default | [1136](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TooltipMinimalPerf.default.html) | [1134](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TooltipMinimalPerf.default_base.html) | 1:1
AccordionMinimalPerf.default | [168](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AccordionMinimalPerf.default.html) | [170](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/AccordionMinimalPerf.default_base.html) | 0.99:1
ChatWithPopoverPerf.default | [399](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ChatWithPopoverPerf.default.html) | [403](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ChatWithPopoverPerf.default_base.html) | 0.99:1
CheckboxMinimalPerf.default | [3043](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/CheckboxMinimalPerf.default.html) | [3059](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/CheckboxMinimalPerf.default_base.html) | 0.99:1
EmbedMinimalPerf.default | [4567](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/EmbedMinimalPerf.default.html) | [4592](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/EmbedMinimalPerf.default_base.html) | 0.99:1
HeaderMinimalPerf.default | [419](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/HeaderMinimalPerf.default.html) | [423](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/HeaderMinimalPerf.default_base.html) | 0.99:1
InputMinimalPerf.default | [1402](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/InputMinimalPerf.default.html) | [1423](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/InputMinimalPerf.default_base.html) | 0.99:1
ListWith60ListItems.default | [710](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ListWith60ListItems.default.html) | [718](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ListWith60ListItems.default_base.html) | 0.99:1
MenuMinimalPerf.default | [948](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/MenuMinimalPerf.default.html) | [960](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/MenuMinimalPerf.default_base.html) | 0.99:1
ProviderMergeThemesPerf.default | [1805](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ProviderMergeThemesPerf.default.html) | [1827](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ProviderMergeThemesPerf.default_base.html) | 0.99:1
IconMinimalPerf.default | [692](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/IconMinimalPerf.default.html) | [697](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/IconMinimalPerf.default_base.html) | 0.99:1
DatepickerMinimalPerf.default | [5952](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/DatepickerMinimalPerf.default.html) | [6089](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/DatepickerMinimalPerf.default_base.html) | 0.98:1
ListMinimalPerf.default | [579](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ListMinimalPerf.default.html) | [591](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/ListMinimalPerf.default_base.html) | 0.98:1
SliderMinimalPerf.default | [1680](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/SliderMinimalPerf.default.html) | [1729](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/SliderMinimalPerf.default_base.html) | 0.97:1
TextAreaMinimalPerf.default | [580](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TextAreaMinimalPerf.default.html) | [596](https://fluentuipr.z22.web.core.windows.net/pull/19186/perf-test-northstar/TextAreaMinimalPerf.default_base.html) | 0.97:1

</details><div id="perfComment9423"/>

behowell commented on packages/react-hooks/src/useRefEffect.ts:
```diff
@@ -35,31 +35,34 @@ export function useRefEffect<T>(callback: (value: T) => (() => void) | void, ini
-          if (data.cleanup) {
-            data.cleanup();
-            data.cleanup = undefined;
+  const cleanup = React.useRef<RefData['cleanup']>();
+  const [ref] = React.useState(() => ({
```
I don't think you need both the `useRef` and `useState`. The `cleanup` value could be a property on the `ref` object instead of a separate `useRef`.

And then replace `cleanup.current` with `ref.cleanup`

```suggestion
  const [ref] = React.useState(() => ({
    cleanup: undefined as (() => void) | void | undefined,
```

behowell commented on packages/react-hooks/src/useRefEffect.ts:
```diff
@@ -35,31 +35,34 @@ export function useRefEffect<T>(callback: (value: T) => (() => void) | void, ini
     callback: (value: T) => (() => void) | void;
     cleanup?: (() => void) | void;
```
Remove `RefData` since you're not really using it anymore. You can explicitly type `cleanup` below.

behowell commented on packages/react-hooks/src/useRefEffect.ts:
```diff
@@ -35,31 +35,34 @@ export function useRefEffect<T>(callback: (value: T) => (() => void) | void, ini
 
-  return data.ref;
+  ref.callback = callback;
+  return (ref.facade as unknown) as RefCallback<T>;
```
This is fishy. `facade` is _not_ a `RefCallback` (a function that takes a new value); it's an object with a get/set function for `current`. useRefEffect should be updated to return `MutableRefObject` instead. You'd need to update the places it's used as well...

behowell reviewed (changes requested):
This changes the API for the object returned by useRefEffect. You'll need to update the signature and the places it's used.

... or make a smaller change that just removes the use of `Object.assign`

ling1726 commented:
> This changes the API for the object returned by useRefEffect. You'll need to update the signature and the places it's used.
> 
> ... or make a smaller change that just removes the use of `Object.assign`

tbh the ref callback thingy is even more dodgy... especially as it's assining a `current` value to the object prototype... I just went ahead and reverted and removed the object assign. But it looks even weirder now.

Feel free to take over the PR if you want
